In [1]:
import scipy.stats as stats
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

In [30]:
df=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

In [31]:
### using the given data, the sample mean(x_bar) and std_dev are calculated
x_bar=np.mean(df)
x_bar

np.float64(1.2386666666666666)

In [32]:
sample_std_dev= np.std(df)
sample_std_dev

np.float64(0.18661427836285438)

## T-Test

To perform t-test,
we need to have 2 values:
1.degree of freedom
2.alpha/2

In [ ]:
### using t-table we can find the value od t-critival as 2.97 (or) we can use stats.t.ppf()

In [43]:
## area=(CI+alpha/2)=0.99+0.005=0.995
t_critical=stats.t.ppf(0.995,14)

In [44]:
### calculating the margin of error
std_error = sample_std_dev / np.sqrt(n)
std_error

np.float64(0.04818359948425667)

In [46]:
mar_of_error = t_critical * std_error
mar_of_error

np.float64(0.14343499804054372)

In [48]:
### finding the lower and upper bound values so as to build the confidence interval
lower_value=(x_bar-mar_of_error)
upper_value=(x_bar+mar_of_error)
print(lower_value,upper_value)


1.095231668626123 1.3821016647072102


#### Hence we can conclude that ,99% confidence interval for the mean durability of the print-heads is (1.090, 1.387) million characters

In [67]:
pop_std_dev=0.2

#### As the standard deviation of the population is known we can use the z-test to find the critical value for 99% CI

In [68]:
## we can find z_critical value using scipy package
from scipy.stats import norm

In [69]:
z_critical=stats.norm.ppf(0.995)
z_critical

np.float64(2.5758293035489004)

In [70]:
### next we find margin of error for using z_critical value

In [71]:
marginal_error=z_critical*(pop_std_dev/np.sqrt(n))
marginal_error

np.float64(0.13301525327090588)

In [72]:
lower_value=(x_bar-marginal_error)
upper_value=(x_bar+marginal_error)
print(lower_value,upper_value)


1.1056514133957607 1.3716819199375725


#### Therefore, we can say that 99% confidence interval for the mean durability of the print-heads is approximately (1.106, 1.372) million characters printed before failure.
.